zebra-scan/src/config.rs

```diff
@@ -0,0 +1,21 @@
+//! Configuration for blockchain scanning tasks.
+
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields, default)]
+/// Configuration for scanning.
+pub struct Config {
+    /// The sapling keys to scan for and the birthday height of each of them.
+    // TODO: any value below sapling activation as the birthday height should default to sapling activation.
+    pub sapling_keys_to_scan: IndexMap<String, u32>,
```

Comment:
Optional:
- Is there any reason why this is `IndexMap` rather than `HashMap`?
- Can we use `zebra_scan::SaplingScanningKey` here?
- It might be less confusing to use the same variable name as the storage, but it really doesn't matter

```rust
    /// The sapling keys to scan for and the birthday height of each of them.
    // TODO: any value below sapling activation as the birthday height should default to sapling activation.
    //       use SaplingScanningKey and Height here
    pub sapling_scanning_keys: HashMap<String, u32>,
```

Reply:
Yea, the only reason i saw is that `IndexMap` guarantees the order of the list which can be useful if we make tests.

I am happy to change it if you think it do not worth it.

Reply:
That seems like a good reason, so let's keep `IndexMap` and say that in the field docs. So we don't accidentally "simplify" it later.

It will also guarantee the order if we read and then write the config, or if there are a large number of keys and we choose some keys to scan first.

Reply:
> Can we use zebra_scan::SaplingScanningKey here?

Ill add this ass soon as https://github.com/ZcashFoundation/zebra/pull/7942 gets merged.